src/Rules/Deprecations/OverrideDeprecatedConstantRule.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassConst;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use function sprintf;
+
+/**
+ * @implements Rule<ClassConst>
+ */
+class OverrideDeprecatedConstantRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return ClassConst::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+			return [];
+		}
+
+		if (!$scope->isInClass()) {
+			return [];
+		}
+
+		if ($node->isPrivate()) {
+			return [];
+		}
+
+		$class = $scope->getClassReflection();
+
+		$parents = $class->getParents();
```

Comment:
You shouldn't go through all the parents. This would break the `@deprecated` /  `@not-deprecated` behaviour. Please read about it and write a test: https://phpstan.org/writing-php-code/phpdocs-basics#deprecations

It's sufficient to ask for the property in the first parent class if there's one.

You should also look into interfaces.